src/commands/bans/ban.ts

```diff
@@ -18,8 +18,12 @@ const ban: Command = {
                 reply = "You can only ban one word at a time";
             } else {
                 const [word] = args;
-                await banImpl(word);
-                reply = `Banned \`${word.toLowerCase()}\``;
+                try {
+                    await banImpl(word);
+                    reply = `Banned \`${word.toLowerCase()}\``;
+                } catch (e) {
+                    reply = e;
```

Comment:
I'm not 100% sure that e will actually be a string and not an Error. If it is the latter, I am not sure how to get the string back out of it

Reply:
#If it's something of type `Error`, it should have a `message` property that you can access

Reply:
Maybe something like
```typescript
} catch (err) {
    Log.err(`Failed to ban "${word}". Reason:`, err);
    reply = err?.message ?? `Can't ban \`${word}\``;
}
```
It tries to pull the error message out and if it doesn't exist it has a default message.

Because you can throw _anything_ in js you can't assume anything about the shape/type of the error.

`?.` means dereference if the pointer is not null/undefined, otherwise evaluate to undefined overall. You might have seen this in other languages like ruby as `foo&.bar`

`??` means if the LHS is defined, use LHS, else use RHS

Reply:
I'm not sure how happy I am with being able to throw anything. I would have thought typescript would have disallowed that. Anyway, I've implemented that suggestion. I feel like we don't really need a log here since having an exception is an intended possibility. I also decided not to print out the word since we are throwing the exception in cases where the word may not be safe to print.

I like this ? and ?? syntax. I feel like I will be using them a lot.